core/persistence-api/src/main/java/org/apache/syncope/core/persistence/api/entity/user/User.java

```diff
@@ -54,8 +55,14 @@
     void setSecurityQuestion(SecurityQuestion securityQuestion);
 
     String getSecurityAnswer();
+    
+    String getClearSecurityAnswer();
 
-    void setSecurityAnswer(String securityAnswer);
+    void setEncodedSecurityAnswer(String securityAnswer);
+    
+    void setSecurityAnswer(String securityAnswer,  CipherAlgorithm cipherAlgoritm);
+
+    boolean canDecodeSecurityAnswer();
```

Comment:
Is this method really needed? Isn't just a copy of `canDecodePassword()`?
We might unify the two methods in one `canDecodeSecrets()`

Reply:
Sure. That's much better